src/views/Map/AgentMAp.js

```diff
@@ -0,0 +1,174 @@
+
+// var map, infoWindow, marker, markers, geoCodeValue;
+var map, pos, panorama, infoWindow;
+
+class AgentMap extends React.PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.googleMapRef = React.createRef()
+    this.initMap = this.initMap.bind(this)
+    this.addRoute = this.addRoute.bind(this)
+    this.handleLocationError = this.handleLocationError.bind(this)
+    this.processSVData = this.processSVData.bind(this)
+  }
+
+  componentDidMount() {
+    let googleScript = document.createElement('script');
+    // API key invalid, add yours or contact @johnerry on git
+    googleScript.src = "https://maps.googleapis.com/maps/api/js?key=AIzatSyBKHtdFymwut91lTBTQBDgltrFaXPD6TxY2g"
+    googleScript.async = true
+    window.document.body.appendChild(googleScript)
+    googleScript.addEventListener('load', this.initMap)
+  }
+
+  initMap() {
+    var google = window.google;
+
+    let addRoute = this.addRoute
+    let handleLocationError = this.handleLocationError
+    let processSVData = this.processSVData
+
+    var latlong = { lat: 6.50954222424663, lng: 3.370920970651687 };
+    var sv = new google.maps.StreetViewService();
+    panorama = new google.maps.StreetViewPanorama(document.getElementById('pano'));
+    // Set up the map.
+    map = new google.maps.Map(document.getElementById('map'), {
+      center: latlong,
+      zoom: 15,
+      streetViewControl: false
+    });
+
+    infoWindow = new google.maps.InfoWindow();
+    // Try HTML5 geolocation.
+    if (navigator.geolocation) {
+      pos = { lat: 6.60954222424663, lng: 3.370920970651687 }
+      navigator.geolocation.getCurrentPosition(function (position) {
+        pos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        if (pos !== undefined) {
+          console.log(pos);
+          addRoute();
+        }
+      }, function () {
+        addRoute()
+        handleLocationError(true, infoWindow, map.getCenter());
+      });
+    } else {
+      addRoute()
+      // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+    }
+    // Set the initial Street View camera to the center of the map
+    sv.getPanorama({ location: null, radius: 50 }, processSVData);
+    map.addListener('click', function (event) {
+      sv.getPanorama({ location: event.latLng, radius: 50 }, processSVData);
+    });
+
+  }
+
+  addRoute() {
+    var google = window.google;
+
+    var directionsRenderer = new google.maps.DirectionsRenderer();
+    var directionsService = new google.maps.DirectionsService();
+    directionsRenderer.setMap(map);
+    directionsRenderer.setPanel(document.getElementById('right-panel'));
+    if (pos === undefined) {
+      var start = { lat: 6.60157422424999, lng: 3.360932070651222 };
+    } else {
+      start = pos;
+    }
+    var end = { lat: 6.60157222424999, lng: 3.360920570651222 };
+    directionsService.route({
+      origin: start,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['start' used outside of binding context.](https://app.codacy.com/gh/BuildForSDG/Team-064-Frontend/pullRequest?prid=5703258)